bark/world/evaluation/rss_interface.cpp

```diff
@@ -0,0 +1,481 @@
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+bool RssInterface::initializeOpenDriveMap(
+    const std::string& opendrive_file_name) {
+  std::ifstream opendrive_file(opendrive_file_name);
+  std::string opendrive_file_content =
+      std::string{std::istreambuf_iterator<char>(opendrive_file),
+                  std::istreambuf_iterator<char>()};
+  ::ad::map::access::cleanup();
+
+  // the 2nd argument is the value of narrowing overlapping between two lanes,
+  // it is only relevent if the map has intersection
+  bool result = ::ad::map::access::initFromOpenDriveContent(
+      opendrive_file_content, 0.01,
+      ::ad::map::intersection::IntersectionType::TrafficLight,
+      ::ad::map::landmark::TrafficLightType::UNKNOWN);
+
+  if (result == false)
+    LOG(ERROR) << "RSS failed to initialize from OpenDrive map : "
+               << opendrive_file_content << std::endl;
+
+  return result;
+}
+
+::ad::rss::world::RssDynamics RssInterface::GenerateVehicleDynamicsParameters(
+    float lon_max_accel, float lon_max_brake, float lon_min_brake,
+    float lon_min_brake_correct, float lat_max_accel, float lat_min_brake,
+    float lat_fluctuation_margin, float response_time) {
+  ::ad::rss::world::RssDynamics dynamics;
+
+  // RSS dynamics values along longitudinal coordinate system axis
+  dynamics.alphaLon.accelMax = Acceleration(lon_max_accel);
+  dynamics.alphaLon.brakeMax = Acceleration(lon_max_brake);
+  dynamics.alphaLon.brakeMin = Acceleration(lon_min_brake);
+  dynamics.alphaLon.brakeMinCorrect = Acceleration(lon_min_brake_correct);
+
+  // RSS dynamics values along lateral coordinate system axis
+  dynamics.alphaLat.accelMax = Acceleration(lat_max_accel);
+  dynamics.alphaLat.brakeMin = Acceleration(lat_min_brake);
+  dynamics.lateralFluctuationMargin = Distance(lat_fluctuation_margin);
+
+  dynamics.responseTime = Duration(response_time);
+
+  // new parameters after ad-rss v4.0.0
+  // leave them as the default one for now
+  dynamics.unstructuredSettings.pedestrianTurningRadius =
+      ad::physics::Distance(2.0);
+  dynamics.unstructuredSettings.driveAwayMaxAngle = ad::physics::Angle(2.4);
+  dynamics.unstructuredSettings.vehicleYawRateChange =
+      ad::physics::AngularAcceleration(0.3);
+  dynamics.unstructuredSettings.vehicleMinRadius = ad::physics::Distance(3.5);
+  dynamics.unstructuredSettings.vehicleTrajectoryCalculationStep =
+      ad::physics::Duration(0.2);
+
+  return dynamics;
+}
+
+::ad::rss::world::RssDynamics RssInterface::GenerateAgentDynamicsParameters(
+    const AgentId& agent_id) {
+  if (agents_dynamics_.find(agent_id) != agents_dynamics_.end()) {
+    return GenerateVehicleDynamicsParameters(
+        agents_dynamics_[agent_id][0], agents_dynamics_[agent_id][1],
+        agents_dynamics_[agent_id][2], agents_dynamics_[agent_id][3],
+        agents_dynamics_[agent_id][4], agents_dynamics_[agent_id][5],
+        agents_dynamics_[agent_id][6], agents_dynamics_[agent_id][7]);
+  } else {
```

Comment:
else is not required. then we have a default case anyway

Reply:
Done it commit https://github.com/bark-simulator/bark/commit/2499d855e7b362aa10d3a319743d6378fd081a0e#diff-ee49029f58331d10b8301038732c1634R84